drivers/led/lp50xx.c

```diff
@@ -302,28 +323,22 @@ static int lp50xx_init(const struct device *dev)
 		}
 	}
 
-	/* Enable hardware */
-	err = lp50xx_hw_enable(dev, true);
-	if (err < 0) {
-		LOG_ERR("%s: failed to enable hardware", dev->name);
-		return err;
-	}
-
-	/* Reset device */
-	err = lp50xx_reset(dev);
-	if (err < 0) {
-		LOG_ERR("%s: failed to reset", dev->name);
-		return err;
+	/* If the device is behind a power domain, it will start in
+	 * PM_DEVICE_STATE_OFF.
+	 */
+	if (pm_device_on_power_domain(dev)) {
```

Comment:
`pm_device_on_power_domain` says nothing about whether the power domain is currently powered or not.